src/main/java/com/uid2/client/PublisherUid2Helper.java

```diff
@@ -105,7 +119,29 @@ private static String decrypt(String response, byte[] secretKey, boolean isRefre
         return new String(resultBytes, StandardCharsets.UTF_8);
     }
 
+    private static String setArtifactAndVersion() {
+        String artifactAndVersion;
+
+        try { // https://stackoverflow.com/a/3697482/297451
+            Class<?> cls = Class.forName("com.uid2.client.PublisherUid2Client");
+
+            InputStream istream = cls.getClassLoader().getResourceAsStream("project.properties");
+            if (istream == null) {
+                throw new IOException("project.properties not found");
+            }
+
+            final Properties properties = new Properties();
+            properties.load(istream);
+            artifactAndVersion = properties.getProperty("artifactId") + "-" + properties.getProperty("version");
+        } catch (ClassNotFoundException | IOException e) {
+            artifactAndVersion = "<unknown - " + e + ">";
```

Comment:
What will "e" expand to when concatenated with a string? Will it create different messages that breaks our prometheus metrics with high cardinality (because there can be so many possible errors)?

Reply:
I think there are only 2 possible error results:
<unknown - java.io.IOException: project.properties not found>
<unknown - java.lang.ClassNotFoundException: com.uid2.client.PublisherUid2Client>

I wouldn't expect to see these in prod unless some package/deployment issue

Reply:
If there are just two can we shorten these a bit, so that when we view it on grafana we can quickly identify them? This is probably NIT.

Reply:
ok, will remove the "unknown - " prefix